mesonbuild/utils/pkgconfig.py

```diff
@@ -0,0 +1,396 @@
+        # "UNIX98 /bin/sh" format. Using mesonlib.split_args() here would make
+        # CI fail on mingw64.
+        return shlex.split(value)
+
+    def _prepend_sysroot(self, path: str) -> str:
+        if not self.sysroot:
+            return path
+        p = Path(path)
+        p = p.relative_to(p.anchor)
+        return Path(self.sysroot, p).as_posix()
+
+    def _eval(self, content: str, variables: T.Dict[str, str]) -> str:
+        dollar = False
+        brackets = False
+        start = 0
+        line = ''
+        for i, c in enumerate(content):
+            if brackets:
+                if c == '}':
+                    brackets = False
+                    varname = content[start:i]
+                    try:
+                        line += variables[varname]
+                    except KeyError:
+                        raise PkgConfigException(f'Variable {varname!r} not defined in {self.pkgname!r}')
+                    start = i + 1
+            elif dollar:
+                if c == '{':
+                    dollar = False
+                    brackets = True
+                    line += content[start:i-1]
+                    start = i + 1
+                elif c == '$':
+                    line += content[start:i]
+                    start = i + 1
+                    dollar = False
+            elif c == '$':
+                dollar = True
+        line += content[start:]
+        return line
+
+    @lru_cache(maxsize=None)
+    def _eval_variables(self, define_variable: PkgConfigDefineType) -> T.Dict[str, str]:
+        variables = self.globals.copy()
+        define_key, define_value = define_variable or (None, None)
+        variables[define_key] = define_value
+        for key, raw_value in self.raw_variables.items():
+            if key == define_key:
+                continue
+            variables[key] = self._eval(raw_value, variables)
+        return variables
+
+    def _eval_field(self, field: str, define_variable: PkgConfigDefineType = None) -> str:
+        variables = self._eval_variables(define_variable)
+        return self._eval(self.raw_fields.get(field, ''), variables)
+
+    def _eval_args(self, field: str, system_args: T.List[str], allow_system_args: bool,
+                   define_variable: PkgConfigDefineType) -> T.List[str]:
+        args = self._parse_args(self._eval_field(field, define_variable))
+        if allow_system_args:
+            return args.copy()
+        return [a for a in args if a not in system_args]
+
+    def _parse_args(self, value: str) -> T.List[str]:
+        args = []
+        for a in self._split_args(value):
+            # Canonicalize '-I /foo' to '-I/foo'
+            if a.startswith('-I') or a.startswith('-L'):
+                a = a[:2] + a[2:].lstrip()
+            args.append(a)
+        return args
+
+    def _parse_requires(self, content: str) -> T.List[RequiredVersion]:
+        SEPARATORS = ' ,'
+        OPERATORS = '<>=!'
```

Comment:
Since these are only being used for `in` checks, a set would be slightly faster.